Modules/_ssl.c

```diff
@@ -553,10 +568,21 @@ newPySSLSocket(PySSLContext *sslctx, PySocketSockObject *sock,
     self->peer_cert = NULL;
     self->ssl = NULL;
     self->Socket = NULL;
-    self->ssl_sock = NULL;
     self->ctx = sslctx;
     self->shutdown_seen_zero = 0;
     self->handshake_done = 0;
+    self->owner = NULL;
+    if (server_hostname != NULL) {
+        hostname = PyUnicode_Decode(server_hostname, strlen(server_hostname),
+                                   "idna", "strict");
```

Comment:
Hum, this change doesn't seem to be directely related to MemoryBIO, or did I miss something?